src/opencl_common.h

```diff
@@ -337,6 +337,22 @@ void opencl_process_event(void);
 #define GET_EXACT_MULTIPLE(dividend, divisor)	  \
 	(divisor) ? ((dividend > divisor) ? ((dividend / divisor) * divisor) : divisor) : dividend
 
+#define WAIT2_INIT(sleep_event)	  \
+	if (opencl_avoid_busy_wait[gpu_id] && !sleep_event) { \
+		opencl_sleep_pid = getpid(); \
+		sleep_event = &opencl_sleep_event; \
+	}
+
+#define WAIT2_SLEEP(sleep_event)	  \
+	if (sleep_event == &opencl_sleep_event) { \
+		BENCH_CLERROR(clSetEventCallback(opencl_sleep_event, CL_COMPLETE, &opencl_event_callback, NULL), "set callback"); \
+		opencl_sleep(); \
+	}
```

Comment:
Note to self: Need to drop BENCH_CLERROR here, and instead only sleep if `clSetEventCallback` succeeds.